### DISCUSSION TITLE
Input pre-generated images and metadata.

### DISCUSSION THREAD

**Discussion by mattyung23** [mcmonkeyprojects/sd-infinity-grid-generator-script#117] [General]
Is there a way to pass in my own generated images and associated parameter settings rather than to have this tool generate the images for me before creating the grid?

**Comment by mcmonkey4eva** [comment]
Depends what you're trying to do?

## 1: Manually Fill In Image Slots

Run the gridgen once, let it get started, then hit interrupt - it now has the base files laid out, and you can go store images in the relevant slots, then can also run the gridgen again if you need it to fill in remaining slots (it won't overwrite images that are already stored).

## 2: Manually make your own grid that doesn't fit the actual standard multi-axis grid format exactly (ie just browse images)

Might want to go a different route, or just use the html/js files as a starting point and customize it. Might consider other tools like [StableSwarmUI](https://github.com/Stability-AI/StableSwarmUI)'s built in image browser

## 3: Writing your own software that builds grids

All the code is open source and relatively straightforward, with a reference implementation in both [python](https://github.com/mcmonkeyprojects/sd-infinity-grid-generator-script/blob/master/gridgencore.py) and [C#](https://github.com/Stability-AI/StableSwarmUI/tree/master/src/BuiltinExtensions/GridGenerator), you're welcome to copy out either reference and use it.

## 4: Something else

If none of the above is what you're trying, please clarify what you're wanting to do
